web-app/js/smartR/_angular/directives/fetchButton.js

```diff
@@ -8,6 +8,8 @@ window.smartRApp.directive('fetchButton',
         return {
             restrict: 'E',
             scope: {
+                loaded: '=',
```

Comment:
I'm not sure about this. The dangerous situation would be, would controller can trust only this 'loaded' flag while the content could be empty.